src/logcollector/logcollector.c

```diff
@@ -152,6 +152,17 @@ void LogCollectorStart()
                        ARGV0);
             }
         }
+        else if (strcmp(logff[i].logformat, "journald") == 0) {
+#ifdef HAVE_SYSTEMD
+            verbose(READING_JOURNAL, ARGV0, logff[i].file);
+            sd_read_journal(logff[i].file);
```

Comment:
Just wondering is this correct way of handling this as other reader just make logff[i].read = sd_read_something(with args) and journald reader hangs forever?